google/cloud/internal/future_coroutines.h

```diff
@@ -0,0 +1,131 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_INTERNAL_FUTURE_COROUTINES_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_INTERNAL_FUTURE_COROUTINES_H
+
+#include "google/cloud/internal/port_platform.h"
+#if GOOGLE_CLOUD_CPP_CPP_VERSION >= 202002L && __cpp_impl_coroutine
+#include "google/cloud/future_generic.h"
+#include "google/cloud/future_void.h"
+#include "google/cloud/version.h"
+#include <coroutine>
+#include <memory>
+
+/// Specialize `std::coroutine_traits` for `google::cloud::future<T>`.
+template <typename T, typename... Args>
+requires(!std::is_void_v<T> && !std::is_reference_v<T>) /**/
```

Comment:
Is `/**/` intentional?

Reply:
yes, `clang-format` does a poor job with these otherwise.